tests/helpers/components.ts

```diff
@@ -0,0 +1,303 @@
+import { createInMemoryIPFS } from "./in-memory-ipfs.js"
+
+
+// 🚀
+
+
+export const configuration: Configuration = {
+  appInfo: { name: "Webnative Tests", creator: "Fission" },
+  debug: false,
+  filesystem: {
+    loadImmediately: false
+  },
+}
+
+
+
+// CRYPTO
+
+
+export async function createCryptoComponent(): Promise<Crypto.Implementation> {
+  const cfg = KeystoreConfig.normalize({
+    type: CryptoSystem.RSA,
+
+    charSize: 8,
+    hashAlg: HashAlg.SHA_256,
+    storeName: "tests",
+    exchangeKeyName: "exchange-key",
+    writeKeyName: "write-key",
+  })
+
+  const { rsaSize, hashAlg, storeName, exchangeKeyName, writeKeyName } = cfg
+  const store = new LocalForageStore() as unknown as LocalForage
+
+  // NOTE: This would be a more type safe solution,
+  //       but somehow localforage won't accept the driver.
+  // await store.defineDriver(memoryDriver)
+  // const store = localforage.createInstance({ name: storeName, driver: memoryDriver._driver })
+
+  await IDB.createIfDoesNotExist(exchangeKeyName, () => (
+    RSAKeys.makeKeypair(rsaSize, hashAlg, KeyUse.Exchange)
+  ), store)
+  await IDB.createIfDoesNotExist(writeKeyName, () => (
+    RSAKeys.makeKeypair(rsaSize, hashAlg, KeyUse.Write)
+  ), store)
+
+  const ks = new RSAKeyStore(cfg, store)
+
+  return {
+    aes: {
+      decrypt: BrowserCrypto.aesDecrypt,
+      encrypt: BrowserCrypto.aesEncrypt,
+      exportKey: BrowserCrypto.aesExportKey,
+      genKey: BrowserCrypto.aesGenKey,
+    },
+    ed25519: {
+      verify: BrowserCrypto.ed25519Verify
+    },
+    hash: {
+      sha256: BrowserCrypto.sha256,
+    },
+    keystore: {
+      clearStore: () => BrowserCrypto.ksClearStore(ks),
+      decrypt: (...args) => BrowserCrypto.ksDecrypt(ks, ...args),
+      exportSymmKey: (...args) => BrowserCrypto.ksExportSymmKey(ks, ...args),
+      getAlg: (...args) => BrowserCrypto.ksGetAlg(ks, ...args),
+      getUcanAlg: (...args) => BrowserCrypto.ksGetUcanAlg(ks, ...args),
+      importSymmKey: (...args) => BrowserCrypto.ksImportSymmKey(ks, ...args),
+      keyExists: (...args) => BrowserCrypto.ksKeyExists(ks, ...args),
+      publicExchangeKey: (...args) => BrowserCrypto.ksPublicExchangeKey(ks, ...args),
+      publicWriteKey: (...args) => BrowserCrypto.ksPublicWriteKey(ks, ...args),
+      sign: (...args) => BrowserCrypto.ksSign(ks, ...args),
+    },
+    misc: {
+      randomNumbers: BrowserCrypto.randomNumbers,
+    },
+    rsa: {
+      decrypt: BrowserCrypto.rsaDecrypt,
+      encrypt: BrowserCrypto.rsaEncrypt,
+      exportPublicKey: BrowserCrypto.rsaExportPublicKey,
+      genKey: BrowserCrypto.rsaGenKey,
+      verify: BrowserCrypto.rsaVerify
+    },
+  }
+}
+
+
+const crypto = await createCryptoComponent()
+
+
+
+// DEPOT
+
+
+// const [ ipfs, repo ] = await createInMemoryIPFS()
+// const depot: Depot.Implementation = await IpfsDepot.implementation(ipfs, repo)
+
+
+export const inMemoryDepot: Record<string, Uint8Array> = {}
+
+
+const depot: Depot.Implementation = {
```

Comment:
This avoids us having to use js-ipfs in the tests.
Bit hacky I guess because I shove large chunked files inside a single block, but it works.